ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -1048,6 +1054,29 @@ export const WorkspaceEdit = fp.flow(
       this.saveWorkspace();
     }
 
+    private async apiCreateWorkspaceAsync() {
+      let pollTimedOut = false;
+      setTimeout(() => (pollTimedOut = true), CREATE_WORKSPACE_TIMEOUT_MS);
+      let workspaceOp = await workspacesApi().createWorkspaceAsync(
+        this.state.workspace
+      );
+      while (
+        !pollTimedOut &&
+        workspaceOp.status === WorkspaceOperationStatus.PENDING
+      ) {
+        await new Promise((resolve) =>
+          setTimeout(resolve, CREATE_WORKSPACE_INTERVAL_MS)
+        );
```

Comment:
If you import `delay` from subscribable,
```suggestion
        await delay(CREATE_WORKSPACE_INTERVAL_MS);
```

Reply:
nice, wasn't aware of this